src/components/Modal/ModalManager.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import ModalContainer from './ModalContainer/ModalContainer';
+
+const mapState = (state) => ({currentModal : state.modals});
+
+const ModalManager = (props) => {
+  const {currentModal} = props
+  let renderedModal;
+  if (currentModal) {
+    const {modalContent = {}} = currentModal;
+    renderedModal = <ModalContainer content={modalContent} />;
+  }
+
+  return (
+    <div>{renderedModal}</div>
+  )
+}
+
+export default connect(mapState)(ModalManager);
```

Comment:
Logic here can be cleaned up a bunch
```suggestion
import React from 'react';
import { connect } from 'react-redux';

const mapState = (state) => ({currentModal : state.modals});

const ModalManager = ({ currentModal }) => {
  return (<div>{ currentModal ? currentModal.modalContent : null }</div>);
}

export default connect(mapState)(ModalManager);

```

Reply:
saves some bites in memory as well

Reply:
newest push is refactored